react-native/README.md

```diff
@@ -35,10 +41,127 @@ import * as PXBThemes from '@pxblue/react-native-themes';
+    <App />
+</ThemeProvider>
+```
+
+#### Alternate Dark Theme
+
+The alternate dark theme (`blueDarkAlt`) should be applied to select components to give them the desired PX Blue styling. The following components should use the alternate dark theme:
+
+-   Activity Indicator
+-   Appbar
+-   Avatar
+-   Badge
+-   Bottom Navigation
+-   Button (`contained` mode variant)
+-   FAB
+-   ProgressBar
+-   Snackbar
+-   TextInput
+
+![Dark Theme Infographic](https://raw.githubusercontent.com/pxblue/themes/master/react-native/assets/dark-theme-infographic.png)
+
+1. For these components, make sure you are using the darkThemeAlt.
+2. Do not use the darkTheme or these components will render using the incorrect color scheme.
+
+##### One-Off Usage
+
+If you are only using a component from this list once or twice in your application, you can pass the alternate theme directly to the component through the `theme` prop.
+
+```tsx
+import { useTheme } from 'react-native-paper';
+import { blueDarkAlt } from '@pxblue/react-native-themes';
+const theme = useTheme();
+...
+<Badge size={24} visible theme={theme.dark ? blueDarkAlt : {}}></Badge>
+```
+
+##### Component-Based Usage
+
+If you are using components frequently, it's best to create a wrapper component that will handle the alternate theme logic. This will allow you to keep your code more readable and avoid errors with inconsistent theme application.
+
+To do this, you define a wrapper component that acts as a pass-through for all of the default props and adds the theme logic.
+
+```tsx
+import React from 'react';
+import { blueDarkAlt } from '@pxblue/react-native-themes';
+import { SomeComponent as PaperComponent, useTheme } from 'react-native-paper';
+...
+export const SomeComponent: typeof PaperComponent = (props) => {
+    const theme = useTheme(props.theme);
+    return (
+        <PaperComponent
+            {...props}
+            theme={
+                theme.dark
+                ? Object.assign({}, blueDarkAlt, props.theme)
+                : {}
+            }
+        />
+    );
+};
+```
+
+You would then use your custom wrapper component throughout your application instead of using the component directly from React Native Paper:
+
+```tsx
+import { SomeComponent } from './path/to/SomeComponent';
+...
+<SomeComponent {...samePropsAsThePaperComponent} />
+```
+
+The `Button` component is a special case that requires the alternate theme only if you are using the contained mode. The wrapper component for the `Button` should look like:
+
+```tsx
+import React from 'react';
+import { blueDarkAlt } from '@pxblue/react-native-themes';
+import { Button, useTheme } from 'react-native-paper';
+...
+export const MyCustomButton: typeof Button = (props) => {
+    const theme = useTheme(props.theme);
+    return (
+        <Button
+            {...props}
+            theme={
+                props.mode === 'contained' && theme.dark
+                    ? Object.assign({}, blueDarkAlt, props.theme)
```

Comment:
Do we need to use the cloneDeep here too on the darkAlt theme?

Reply:
We don't need it here. The TextInput component is the only one that needs it. I made false assumptions about this component but the bug was just a result of the TextInput issue.